plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.shopper.block_theme.side_effects.spec.ts

```diff
@@ -128,4 +131,151 @@ test.describe( 'Shopper → Cart block', () => {
 			`${ __dirname }/update-price-plugin.php`
 		);
 	} );
+
+	test( 'User can view empty cart message', async ( {
+		frontendUtils,
+		page,
+	} ) => {
+		await frontendUtils.emptyCart();
+		await frontendUtils.goToCart();
+
+		// Verify cart is empty
+		await expect(
+			page.getByRole( 'heading', {
+				name: 'Your cart is currently empty!',
+			} )
+		).toBeVisible();
+	} );
+
+	test( 'User can remove a product from cart', async ( {
+		frontendUtils,
+		page,
+	} ) => {
+		await frontendUtils.goToShop();
+		await frontendUtils.addToCart( SIMPLE_PHYSICAL_PRODUCT_NAME );
+		await frontendUtils.goToCart();
+		await page
+			.getByLabel( `Remove ${ SIMPLE_PHYSICAL_PRODUCT_NAME } from cart` )
+			.click();
+
+		// Verify product is removed from the cart'
+		await expect(
+			page.getByRole( 'heading', {
+				name: 'Your cart is currently empty!',
+			} )
+		).toBeVisible();
+	} );
+
+	test( 'User can update product quantity', async ( {
+		frontendUtils,
+		page,
+	} ) => {
+		await frontendUtils.goToShop();
+		await frontendUtils.addToCart( SIMPLE_VIRTUAL_PRODUCT_NAME );
+		await frontendUtils.goToCart();
+
+		// Via the input field
+		await page
+			.getByLabel(
+				`Quantity of ${ SIMPLE_VIRTUAL_PRODUCT_NAME } in your cart.`
+			)
+			.fill( '4' );
+
+		// Verify the "Proceed to Checkout" button is disabled during network request
+		await expect(
+			page.getByRole( 'button', { name: 'Proceed to Checkout' } )
+		).toBeDisabled();
+
+		// Verify the "Proceed to Checkout" button is enabled after network request
+		await expect(
+			page.getByRole( 'link', { name: 'Proceed to Checkout' } )
+		).toBeEnabled();
+
+		await expect(
+			page.getByLabel(
+				`Quantity of ${ SIMPLE_VIRTUAL_PRODUCT_NAME } in your cart.`
+			)
+		).toHaveValue( '4' );
+
+		// Via the plus button
+		await page
+			.getByLabel(
+				`Increase quantity of ${ SIMPLE_VIRTUAL_PRODUCT_NAME }`
+			)
+			.click();
+		// Verify the "Proceed to Checkout" button is disabled during network request
+		await expect(
+			page.getByRole( 'button', { name: 'Proceed to Checkout' } )
+		).toBeDisabled();
+
+		// Verify the "Proceed to Checkout" button is enabled after network request
+		await expect(
+			page.getByRole( 'link', { name: 'Proceed to Checkout' } )
+		).toBeEnabled();
+
+		await expect(
+			page.getByLabel(
+				`Quantity of ${ SIMPLE_VIRTUAL_PRODUCT_NAME } in your cart.`
+			)
+		).toHaveValue( '4' );
```

Comment:
We increased the quantity at line 203. 
```suggestion
		).toHaveValue( '5' );
```

Reply:
Good catch, @tarunvijwani. I'll commit your suggestion.